### ISSUE TITLE
Hazelcast Jet doesn't respect system property for hazelcast shutdownhook during initialization

### ISSUE THREAD

**Issue by Alexichlol** [hazelcast/hazelcast-jet#1458]
```

        Properties jetProps = jetConfig.getProperties();
        Properties hzProperties = hzConfig.getProperties();

        // Disable HZ shutdown hook, as we will use the Jet-specific property instead
        String hzHookEnabled = hzProperties.getProperty(
                SHUTDOWNHOOK_ENABLED.getName(), SHUTDOWNHOOK_ENABLED.getDefaultValue()
        );
        if (!jetProps.containsKey(JET_SHUTDOWNHOOK_ENABLED)) {
            jetProps.setProperty(JET_SHUTDOWNHOOK_ENABLED.getName(), hzHookEnabled);
        }
        hzConfig.setProperty(SHUTDOWNHOOK_ENABLED.getName(), "false");

        // copy Jet properties to HZ properties
        for (String prop : jetProps.stringPropertyNames()) {
            hzProperties.setProperty(prop, jetProps.getProperty(prop));
        }
```

This part of code in Jet.java never reads from hzConfig doing Config.getProperty so it never gets system property and then overrides the system property for jet shutdown hook and hazelcast shutdown hook so it will never access the system property anymore.

**Comment by cangencer** [comment]
Yes indeed we should be calling `hzConfig.getProperty()` instead of `hzProperties.getProperty()`. Would you like to send a PR to fix it?

**Comment by Alexichlol** [comment]
@cangencer hzConfig.getProperty() doesn't have the default value though. Do you want to do something like Optional.ofNullable(hzConfig.getProperty()).orElse(DEFAULT_VALUE) ?

**Comment by cangencer** [comment]
@Alexichlol Yes that would be fine.

**Comment by Alexichlol** [comment]
@cangencer not familiar with the whole process of contribution to Hazelcast. I guess I would need to sign contribution paper or not for small patch? Nevertheless https://github.com/hazelcast/hazelcast-jet/pull/1461 created PR.

**Comment by cangencer** [comment]
Thanks! Yes we need the CLA signed before we can merge it: https://hazelcast.atlassian.net/wiki/spaces/COM/pages/6357071/Hazelcast+Contributor+Agreement

**Comment by Alexichlol** [comment]
sent

**Comment by cangencer** [comment]
fixed by #1461
